impeller/aiks/canvas.cc

```diff
@@ -308,41 +308,121 @@ void Canvas::DrawPaint(const Paint& paint) {
 }
 
 bool Canvas::AttemptDrawBlurredRRect(const Rect& rect,
-                                     Size corner_radius,
+                                     Size corner_radii,
                                      const Paint& paint) {
   if (paint.color_source.GetType() != ColorSource::Type::kColor ||
       paint.style != Paint::Style::kFill) {
     return false;
   }
 
-  if (!paint.mask_blur_descriptor.has_value() ||
-      paint.mask_blur_descriptor->style != FilterContents::BlurStyle::kNormal) {
+  if (!paint.mask_blur_descriptor.has_value()) {
     return false;
   }
+
   // A blur sigma that is not positive enough should not result in a blur.
   if (paint.mask_blur_descriptor->sigma.sigma <= kEhCloseEnough) {
     return false;
   }
 
-  Paint new_paint = paint;
-
   // For symmetrically mask blurred solid RRects, absorb the mask blur and use
   // a faster SDF approximation.
 
-  auto contents = std::make_shared<SolidRRectBlurContents>();
-  contents->SetColor(new_paint.color);
-  contents->SetSigma(new_paint.mask_blur_descriptor->sigma);
-  contents->SetRRect(rect, corner_radius);
+  Paint rrect_paint = paint;
+
+  // Absorb the color filter, if any.
+  if (rrect_paint.HasColorFilter()) {
+    rrect_paint.color =
+        rrect_paint.GetColorFilter()->GetCPUColorFilterProc()(paint.color);
+    rrect_paint.color_filter = nullptr;
+    rrect_paint.invert_colors = false;
```

Comment:
nit: This seems tricky, having to clear out invert_colors.  I could see someone missing this easily, it might be worth making a function `AbsorbColorFilter`

Reply:
This should be turned around. Rather than clone and eliminate properties, create a default paint and only copy in the attributes you want.

Reply:
I fixed this up to try and make the paint construction less confusing. Please TAL.